### ISSUE TITLE
Is there support for JSX fragments?

### ISSUE THREAD

**Issue by mhluska** [babel/babel-sublime#348]
React fragments have been awesome but they completely mess up the syntax highlighter.

For example:

```jsx
<>
  <span>hello</span>
  <span>world</span>
</>
```

https://github.com/facebook/jsx/issues/84

**Comment by borela** [comment]
https://github.com/borela/naomi

**Comment by gaearon** [comment]
cc @zertosh

**Comment by borela** [comment]
@gaearon Did you find any bugs on FJSX15?

**Comment by gaearon** [comment]
I'm not super happy with how it special-cases `state` or changes the default extension.

**Comment by borela** [comment]
I can remove the special case for state as it had some issues with other color schemes, it's an easy fix. I am not sure what you mean about the default extension, isn't it coloring .js files?

**Comment by borela** [comment]
Ooo I know what you mean, it wasn't using FJSX15 by default, this is something that I was working on the lastest pulls, https://github.com/borela/naomi/releases/tag/v3.4.0 it should be working as expected and I removed the special cases for state.

**Comment by Thom1729** [comment]
@gaearon [JS Custom](https://github.com/Thom1729/Sublime-JS-Custom) is a configurable extension of Sublime's core syntax highlighting that supports JSX. As of fifteen minutes ago, it should be a complete superset of babel-sublime's functionality.

Because babel-sublime is basically defunct, I'm hoping to establish a seamless migration path once I've gone down the babel-sublime bug tracker and made sure all of the issues are fixed in JS Custom, but it should be production-ready right now.

**Comment by suchipi** [comment]
@im-sunil use [JS Custom](https://github.com/Thom1729/Sublime-JS-Custom) instead

**Comment by vadzim-revolist** [comment]
+1

JS Custom has many disadvantages, like not coloring jsx keys, changing colors in functions inside jsx, and coloring object keys as strings when they just have default values. It's not possible to setup it to look like babel-sublime.

**Comment by Thom1729** [comment]
@vadzim JS Custom is intended to function as a drop-in replacement. Could you post examples of the differences you're talking about?

**Comment by vadzim** [comment]
@Thom1729 this is what I remembered:

code:
```javascript
const ComponentA = ({
  valueA,
  valueB = {
    'fieldA': 'a',
    fieldB,
  },
  valueC,
}) => <>
  {`ComponentB${valueC}`}
  <ComponentB
    propA={valueA}
    propB={typeof valueB === 'function' ? valueB(id) : valueB}
  />
</>
```
babel-sublime (locally patched for fragments):
![default](https://user-images.githubusercontent.com/234171/46970933-e2a8ae00-d0ba-11e8-9c38-993d357a6b02.png)

js custom:
![default](https://user-images.githubusercontent.com/234171/46970916-d58bbf00-d0ba-11e8-91ed-87fb55d2b808.png)

config:
```json
{
	"defaults": {
		"string_object_keys": true	
	}
}
```
Differences in js-custom that I guess should be probably eliminated be modifying theme, but present by default:
- brackets in template strings are visually connected with content
- in jsx `key={variable}` expressions has no highlighting
- in jsx variable names inside an expression `propB={typeof valueB === 'function' ? valueB(id) : valueB}` has not a black color as they have outside of jsx

Differences that cannot be eliminated by modifying theme:
- `valueB` has color like `fieldA`, but in babel-sublime it has color like `fieldB`
- different colors of commas in parameter list of a function

**Comment by vadzim** [comment]
@Thom1729 in my sublime that color scheme is named as LAZY. Indeed, js-custom looks better with other color schemes, but babel-sublime still remains easier to use for me because it just works)

I'm not sure I've got your thoughts about arrow functions and what should be filled in that new issue. 

I've found another difference:
code:
```javascript
const { valueA, valueB: varB = { 'fieldA': 'a', fieldB: 'b' }} = {}
```
babel:
![default](https://user-images.githubusercontent.com/234171/46974087-5058d800-d0c3-11e8-90e5-db4ccb9f0574.png)

js-custom:
![default](https://user-images.githubusercontent.com/234171/46974046-3e773500-d0c3-11e8-8414-c487215000f3.png)

note `valueB` color

**Comment by Thom1729** [comment]
> in my sublime that color scheme is named as LAZY.

It looks like that's one of the Sublime 2 legacy color schemes. Unfortunately, it doesn't look like those are maintained anymore, so I would expect them to have problems with newer syntaxes. There may be a third-party update out there somewhere.

> I'm not sure I've got your thoughts about arrow functions and what should be filled in that new issue.

I had a few spare moments, so I submitted a PR to improve the fallback behavior in that case. Hopefully it will be reviewed and merged into the default package soon. Thanks for pointing out the issue.

> `const { valueA, valueB: varB = { 'fieldA': 'a', fieldB: 'b' }} = {}`

This is rooted in another core syntax bug: the destructuring implementation doesn't correctly handle unbound property names that aren't bare identifiers. This is rather embarrassing; we have four hundred seventy-nine assertions in the bindings test and none of them covers this case. I'll take a look at it posthaste.

**Comment by Thom1729** [comment]
I've submitted a PR to implement the missing cases for object destructuring. Once that change is merged into core, JS Custom will need a minor change to extend `string_object_keys`.

**Comment by sparrowDom** [comment]
Thanks for he naomi link. This has been my frustration for far too long

**Comment by anle1337** [comment]
+1 using babel and sublime doesn't highlight  <> </> right :[

**Comment by max-mykhailenko** [comment]
@anle1337 I recommend to use https://packagecontrol.io/packages/JSCustom instead

**Comment by WesUnwin** [comment]
Really wish this would get fixed, still an issue. I guess the only the reason why the PR is not merged, is its missing tests?

**Comment by WesUnwin** [comment]
This is a duplicate issue of https://github.com/babel/babel-sublime/issues/368

**Comment by Thom1729** [comment]
This should be fixed in v10.0.0-beta.1. See #379 for more information on the new release. Please report any problems with v10 in a new issue.
